blobs.go

```diff
@@ -185,6 +190,16 @@ type BlobWriter interface {
 	Reader() (io.ReadCloser, error)
 }
 
+// BlobKeeper combines the operations needed for registry's upkeeping such as
+// blob access, read, enumeration and delete. Can be used to prune unnecessary
+// objects.
+type BlobKeeper interface {
```

Comment:
@miminar Adding `Blobs(...) BlobStore` to `Namespace` seems to make sense. It does bring it dangerously close to `Repository` but it may make sense to unify them.

As far as `BlobKeeper` is concerned, I'd prefer if we used to `BlobService` and `BlobStore` to differentiate Read/Write capability. We we really need here is something that can allow all reads, enumerate and allow deletion. This sounds an awful lot like a BlobService, where as a `BlobStore` would allow ingestion.

Really, the simple thing to do is to merely have a read-restricted `BlobService` and then type assert for enumeration and deletion.